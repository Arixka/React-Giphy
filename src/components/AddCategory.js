import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {

  const [inputValue, setInputValue] = useState('')

  const handleInputValue = (e) => {
    setInputValue(e.target.value)
  }

  const handleSubmit = (e) => {
    console.log('hangleSubmit', inputValue)
    e.preventDefault();
    if(inputValue.trim().length>2){
      // console.log(setCategories(cat=>[inputValue,...cat]))
      setInputValue('')
    }

  }
  return (
    <form onSubmit={handleSubmit}>
      <p>{inputValue}</p>
      <input
        type='text'
        placeholder='Add Category'
        value={inputValue}
        onChange={handleInputValue}
      ></input>
    </form>
  )
}

AddCategory.propTypes= {
  setCategories: PropTypes.func.isRequired
}
