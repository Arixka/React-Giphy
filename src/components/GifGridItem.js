import React from 'react'
import PropTypes from 'prop-types'

export const GifGridItem = ({id, title,url}) => {
 // console.log(id, title,url)
  return (
    <div className='card animate__animated animate__fadeIn animate__delay-2s'>
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  )
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}