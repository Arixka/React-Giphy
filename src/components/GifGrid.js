import React from 'react'
import PropTypes from 'prop-types'
import {useFetchGifs} from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem'

export const GifGrid = ({ category }) => {

  const {data:images, loading} = useFetchGifs(category)

  return (
    <>
      {loading && <p className="animate__animated animate__flash animate__delay-2s">Loading...</p>}

      <h3 className="animate__animated animate__fadeIn animate__delay-2s">
        {category}
      </h3>
      <div className='card-grid'>
        {images.map((img) =>
          <GifGridItem
            key={img.id}
            {...img} />//...img mandamos cada imagen con sus propiedades 
        )}

      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}