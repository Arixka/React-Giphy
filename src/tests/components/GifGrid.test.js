import React from 'react'
import { shallow } from 'enzyme'
import { GifGrid } from '../../components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchGifs'

jest.mock('../../hooks/useFetchGifs') //simular el uso de useFetchGifs
describe('pruebas con gifGrid', () => {

  const category = 'mapaches'
  const gifs = [{ id: 'abc', url: 'http://loquesea', title: 'cualquiercosa' }, { id: 'aeq', url: 'http://loqueseaasa', title: 'cualquiercosaasda' }]

  test('Debe renderizar correctamente el componente', () => {

    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    })
    const wrapper = shallow(<GifGrid category={category} />)
    expect(wrapper).toMatchSnapshot()
  })

  test('Debe de mostrar items cuando cargamos imagenes useFetchGifs', () => {
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    })
    const wrapper = shallow(<GifGrid category={category} />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('p').exists()).toBe(false)
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
  })

})