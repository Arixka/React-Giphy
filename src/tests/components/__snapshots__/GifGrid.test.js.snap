// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas con gifGrid Debe de mostrar items cuando cargamos imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn animate__delay-2s"
  >
    mapaches
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="cualquiercosa"
      url="http://loquesea"
    />
    <GifGridItem
      id="aeq"
      key="aeq"
      title="cualquiercosaasda"
      url="http://loqueseaasa"
    />
  </div>
</Fragment>
`;

exports[`pruebas con gifGrid Debe renderizar correctamente el componente 1`] = `
<Fragment>
  <p
    className="animate__animated animate__flash animate__delay-2s"
  >
    Loading...
  </p>
  <h3
    className="animate__animated animate__fadeIn animate__delay-2s"
  >
    mapaches
  </h3>
  <div
    className="card-grid"
  />
</Fragment>
`;
